stdlib/public/core/HashedCollections.swift.gyb

```diff
@@ -2487,20 +2494,305 @@ internal class _RawNative${Self}Buffer :
+      theState.mutationsPtr = _fastEnumerationStorageMutationsPtr
+    }
+    state.pointee = theState
+    
+    return 0
+  }
+
+%if Self == 'Set':
+
+  @objc
+  internal required init(objects: UnsafePointer<AnyObject?>, count: Int) {
+    _sanityCheckFailure("don't call this designated initializer")
+  }
+
+  @objc
+  internal func member(_ object: AnyObject) -> AnyObject? {
+    return nil
+  }
+
+  @objc
+  internal func objectEnumerator() -> _NSEnumerator {
+    return enumerator()
+  }
+  
+%elif Self == 'Dictionary':
+
+  @objc
+  internal required init(
+    objects: UnsafePointer<AnyObject?>,
+    forKeys: UnsafeRawPointer,
+    count: Int
+  ) {
+    _sanityCheckFailure("don't call this designated initializer")
+  }
+
+  @objc(objectForKey:)
+  func objectFor(_ aKey: AnyObject) -> AnyObject? {
+    return nil
+  }
+
+  func keyEnumerator() -> _NSEnumerator {
+    return enumerator()
+  }
+
+  // We also override the following methods for efficiency.
+
+  func getObjects(_ objects: UnsafeMutablePointer<AnyObject>?,
+    andKeys keys: UnsafeMutablePointer<AnyObject>?) {
+    // Do nothing, we're empty
+  }
+
+%end
+#endif
+}
+
+/// Adds types to a RawNativeBuffer, so that deinit can be implemented.
+/// Doesn't requires Hashable so that it can be instantiated with AnyHashable
+/// in _SwiftDeferredNSSelf for the bridged buffer.
+@_versioned
+internal class _TypedNative${Self}Buffer<${TypeParameters}> :
+  _RawNative${Self}Buffer {
+
+%if Self == 'Set': # Set needs these to keep signatures simple.
+  internal typealias Key = ${TypeParameters}
+  internal typealias Value = ${TypeParameters}
+%end
+
+  deinit {
+    let capacity = _body.capacity
+    let initializedEntries = _UnsafeBitMap(
+        storage: _initializedHashtableEntriesBitMapStorage, bitCount: capacity)
+    let keys = _body.keys.assumingMemoryBound(to: Key.self)
+%if Self == 'Dictionary':
+    let values = _body.values.assumingMemoryBound(to: Value.self)
+%end
+
+    if !_isPOD(Key.self) {
+      for i in 0 ..< capacity {
+        if initializedEntries[i] {
+          (keys+i).deinitialize()
+        }
+      }
+    }
+
+%if Self == 'Dictionary':
+    if !_isPOD(Value.self) {
+      for i in 0 ..< capacity {
+        if initializedEntries[i] {
+          (values+i).deinitialize()
+        }
+      }
+    }
+%end
+    _fixLifetime(self)
+  }
+
   override internal init(_doNotCallMe: ()) {
     _sanityCheckFailure("Only initialize these by calling create")
   }
+
+%if Self == 'Set':
+  @objc
+  internal required init(objects: UnsafePointer<AnyObject?>, count: Int) {
+    _sanityCheckFailure("don't call this designated initializer")
+  }
+%elif Self == 'Dictionary':
+  @objc
+  internal required init(
+    objects: UnsafePointer<AnyObject?>,
+    forKeys: UnsafeRawPointer,
+    count: Int
+  ) {
+    _sanityCheckFailure("don't call this designated initializer")
+  }
+%end
+}
+
+/// Adds the `Key: Hashable` constraint to _NativeTyped
```

Comment:
I don't understand the above comment.

Also, it doesn't follow the API guidelines for doc comments.